### PR TITLE
Display ingress resources without annotations

### PR DESCRIPTION
**Pull request by jasonBirchall**
What does this PR do?
---

This PR connects to https://github.com/ministryofjustice/cloud-platform/issues/3072 and relates to the requirement to list all namespaces, ingress rules and hostnames in live-1 (current live cloud-platform cluster) that don't have an aws weighting annotation.

There are two parts to this PR:
- Backend computation: This is mainly everything in the package main of the reports directory `ingress-weighting`, which actually performs all of the heavy lifting. The package will authenticate to live-1, introspect all ingress resources in the cluster and report on rules that don't contain the required annotation to split traffic between two clusters.
- Front end rendering: This is everything else. I've followed the Ruby Sinatra guide rails and create a simple table for my json data to display both the namespace name and the hostname.

Why merge this PR?
---

Having this data displayed in a table allows us to cross reference future pull requests and highlight when/if a user migrates without making the required ingress change in live-1.

### CODE REVIEW

File: views/ingress_weighting.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    <table class="table">
+  <thead class="thead-dark">
+    <tr>
+      <th scope="col">Namespace</th>
+      <th scope="col">Host</th>
+    </tr>
+  </thead>
+  <tbody>
+  <% list.each do |service| %>
+    <% namespace = service["namespace"] %>
```
Comment (poornima-krishnasamy) [inline views/ingress_weighting.erb:12]: The namespace variable is not used anywhere. so dont need this line

Comment (jasonBirchall) [reply views/ingress_weighting.erb]: Good spot.

Comment (jasonBirchall) [reply views/ingress_weighting.erb]: 21e2907

File: reports/ingress-weighting/main.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+// the rest of the map consists of a primary key (string) with a value containing a map (string:string)
+// of Kubernetes namespaces and hostnames that don't contain the annotation contained within
+// the variable 'annotation'.
+type resourceMap map[string]interface{}
+
+var (
+	hoodawApiKey   = flag.String("hoodawAPIKey", os.Getenv("HOODAW_API_KEY"), "API key to post data to the 'How out of date are we' API")
+	hoodawHost     = flag.String("hoodawHost", os.Getenv("HOODAW_HOST"), "Hostname of the 'How out of date are we' API")
+	bucket         = flag.String("bucket", os.Getenv("KUBECONFIG_S3_BUCKET"), "AWS S3 bucket for kubeconfig")
+	configFile     = flag.String("configFile", "live-1-only", "Name of kubeconfig file in S3 bucket")
```
Comment (poornima-krishnasamy) [inline reports/ingress-weighting/main.go:36]: @jasonBirchall , why do we need a seperate kubeconfig which has only-live-1. Can we not use the existing kubeconfig which concourse has?

Comment (jasonBirchall) [reply reports/ingress-weighting/main.go:36]: It's a great question. The k8s client package isn't brilliant at handling multiple contexts, it adds so much more complexity to the code. I felt this was the easiest win for all.

**Review by poornima-krishnasamy — DISMISSED** [review]
LGTM

**Review by poornima-krishnasamy — APPROVED** [review]

**Pull request merged**